server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageBlobsDAO.java

```diff
@@ -119,20 +123,24 @@
 
     @Override
     public InputStream read(BlobId blobId) throws ObjectStoreException {
-        Blob blob = blobStore.getBlob(containerName.value(), blobId.asString());
+        Optional<Blob> retrievedBlob = Mono.fromCallable(() -> Optional.ofNullable(blobStore.getBlob(containerName.value(), blobId.asString())))
+            .retryBackoff(READ_RETRIES_NUMBER, READ_RETRIES_BACKOFF)
+            .block();
```

Comment:
there is a `blockOptional()` method if I remember well

Reply:
Not really the same, here I want to prevent a null in the Reactor publisher